agent/agent.go

```diff
@@ -491,6 +499,25 @@ func (a *agent) setLifecycle(ctx context.Context, state codersdk.WorkspaceAgentL
 	}
 }
 
+// fetchServiceBannerLoop fetches the service banner on an interval.
+func (a *agent) fetchServiceBannerLoop(ctx context.Context) {
+	ticker := time.NewTicker(adjustIntervalForTests(15*time.Second, time.Millisecond*100))
```

Comment:
Should we increase this to like 2 minutes? 15 seconds seems awfully frequent for something that's not likely to change very often, and if it does there should be quite a bit of heads up (e.g. doing maintenance tomorrow).

Reply:
Really good point.  I think I was overly concerned with someone starting a session right after the banner was updated but like you said there should be plenty of heads up.